adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala

```diff
@@ -503,19 +605,53 @@ case class ReferenceRegion(
   }
 
   /**
+   * Determines if two regions are colocated on the same strand.
+   *
+   * @param other The other region.
+   * @return True if the two are on the same strand, false otherwise
+   */
+  def sameStrand(other: ReferenceRegion): Boolean = {
+    strand == other.strand
+  }
+
+  /**
+   * Determines if two regions are colocated on the same reference name.
+   *
+   * @param other The other region.
+   * @return True if the two are on the same reference name, false otherwise.
+   */
+  def sameReferenceName(other: ReferenceRegion): Boolean = {
+    referenceName == other.referenceName
+  }
+
+  /**
    * @return The length of this region in bases.
    */
   def length(): Long = {
     end - start
   }
 
   override def hashCode: Int = {
-    var result = 37
-    result = 41 * result + (if (referenceName != null) referenceName.hashCode else 0)
-    result = 41 * result + start.hashCode
-    result = 41 * result + end.hashCode
-    result = 41 * result + (if (strand != null) strand.ordinal() else 0)
-    result
+    // add 37 here because this is the start of the hashCode
+    val nameHashCode = 37 + {
+      if (referenceName != null) {
+        referenceName.hashCode
+      } else {
+        0
+      }
+    }
+    val strandHashCode =
+      if (strand != null) {
+        strand.ordinal()
+      } else {
+        0
+      }
+
+    val listOfHashCodes = List(nameHashCode, start.hashCode, end.hashCode, strandHashCode)
```

Comment:
`hashCode` is in general a performance critical operation. As such, allocating and then folding over a list is to be avoided. Why not just write out the multiply/sum chain?